src/secrets/exec.rs

```diff
@@ -0,0 +1,164 @@
+use std::collections::HashMap;
+
+use bytes::BytesMut;
+use futures::executor;
+use futures_util::StreamExt;
+use serde::{Deserialize, Serialize};
+use tokio::{io::AsyncWriteExt, process::Command, time};
+use tokio_util::codec;
+use vector_config::{component::GenerateConfig, configurable_component};
+
+use crate::{config::SecretBackend, signal};
+
+/// Configuration for the `exec` secrets backend.
+#[configurable_component(secrets("exec"))]
+#[derive(Clone, Debug)]
+pub struct ExecBackend {
+    /// Command arguments to execute.
+    ///
+    /// The path to the script or binary must be the first argument.
+    pub command: Vec<String>,
+
+    /// The timeout, in seconds, to wait for the command to complete.
+    #[serde(default = "default_timeout_secs")]
+    pub timeout: u64,
+}
+
+impl GenerateConfig for ExecBackend {
+    fn generate_config() -> toml::Value {
+        let backend = ExecBackend {
+            command: vec![String::from("/path/to/script")],
+            timeout: 3,
+        };
+
+        toml::Value::try_from(&backend).expect("should not fail to serialize")
+    }
+}
```

Comment:
Slightly shorter to [this](https://github.com/vectordotdev/vector/blob/8ec9db3cdbd60d5afdd7bc2561b6ec8a60e2ac80/src/sinks/influxdb/metrics.rs#L143)? Mostly don't think I'd seen the `generate_config` func written quite like that.

(also changed the timeout to match the default)

```suggestion
impl GenerateConfig for ExecBackend {
    fn generate_config() -> toml::Value {
        toml::Value::try_from(Self {
            command: vec![String::from("/path/to/script")],
            timeout: 5,
        })
        .unwrap()
    }
}
```

Reply:
Yeah, fair point. Was thinking I needed to use a reference to a `Serialize` value, but you're right, it can be an owned value constructed inline.

Will fix.

Reply:
Fixed in 8b55d4493.